### PR TITLE
test: default acceptnonstdtxn=0 on all chains

### PR DESCRIPTION
**Pull request by ajtowns**
Changes `-acceptnonstxtxn` to default to 0 on testnet, matching the other chains. Allowing non-standard txs on testnet by default contributed to the difficulties RSK described in #26348: "We see that there are two script paths and, to reduce the script size, a single CHECKMULTISIG is used for the two paths, separating the signer count from the CHECKMULTISIG opcode. This script worked on testnet, because it lacks the standard checks performed in Mainnet."

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -585,7 +585,7 @@ void SetupServerArgs(ArgsManager& argsman)
 
     SetupChainParamsBaseOptions(argsman);
 
-    argsman.AddArg("-acceptnonstdtxn", strprintf("Relay and mine \"non-standard\" transactions (%sdefault: %u)", "testnet/regtest only; ", !testnetChainParams->RequireStandard()), ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::NODE_RELAY);
+    argsman.AddArg("-acceptnonstdtxn", strprintf("Relay and mine \"non-standard\" transactions (%sdefault: %u)", "signet/testnet/regtest only; ", DEFAULT_ACCEPT_NON_STD_TXN), ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::NODE_RELAY);
```
[the remark this answers is not in the document]
Comment (ajtowns) [reply src/init.cpp]: Done

**Comment by instagibbs** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/28354/commits/13eb8aa572644a53ae0d631916cb4cbc273a92d1

**Review by theStack — APPROVED** [review]
Code-review ACK 13eb8aa572644a53ae0d631916cb4cbc273a92d1

**Review by glozow — COMMENTED** [review]
ACK 13eb8aa572644a53ae0d631916cb4cbc273a92d1

**Review by jonatack — COMMENTED** [review]
Post-merge ACK.

**Pull request merged**